[WIP] Added .travis.yml

bsegault opened the pull request.

fuhrmannb commented on .travis.yml:
```diff
@@ -0,0 +1,13 @@
+   - go get -u -v github.com/spf13/hugo
+   - git config --global user.name "Travis CI by SBC IT Team"
+   - git config --global user.email "it@starburst-computing.com"
+script:
```
This script pushing the website must only be executed at the PR merging, not before.
You have to execute it only at this moment, not at the PR creation.

At the PR creation, only site generation must be checked.

bsegault replied:
The `id_rsa.enc` was created by following a tutorial where travis would push content to a repo using ssh. I don't understand exactly how it works yet ^^'

And I understand what you say with the pushing only occuring when merging, but for tests it is way easier for now to do it this way ^^'

fuhrmannb commented on .gitmodules:
```diff
@@ -1,4 +0,0 @@
-[submodule "public"]
```
Why removing the submodule solution?
The previous solution appeared pretty clean, with another repository containing the generated sources of the website... =/

bsegault replied:
I am only trying options here.
I just thought for a second that having to repose for the website maybe a bit intricated. But in a way, we could have on private with hugo source code and one public with only the static-generated content, if you prefer :-)

bsegault commented:
Update @fuhrmannb I updated most of the script and now there is:
 - Distinct repo for source code (this one) and generated code (@Starurstcomputing/starburstcomputing.github.io used for github pages)
 - Removed useless files such as `id_rsa.enc` (replaced by `travis.key.enc` which is used to push to the github pages repo

bsegault commented:
@fuhrmannb I forgot to mention that I still need to push only at merging. Do you know how to do that? Because I didn't find anything on this :-/

fuhrmannb commented:
@bsegault Maybe you can find something in this posts: [here](http://speps.github.io/articles/hugo-setup/) and [here](http://stackoverflow.com/questions/31882306/how-to-configure-travis-ci-to-build-pull-requests-merges-to-master-w-o-redunda).
The idea is to activate the build only on master branch pushes, that are done on PR merges.
